packages/fast-form-generator-react/app/index.tsx

```diff
@@ -52,54 +59,86 @@ export default class App extends React.Component<{}, IAppState> {
      */
     private childOptions: IChildOptionItem[];
 
-    /**
-     * The schema form attribute settings mapping configuration
-     */
-    private attributeSettingsMappingToPropertyNames: IFormAttributeSettingsMappingToPropertyNames;
-
     constructor(props: {}) {
         super(props);
 
         this.childOptions = this.getChildOptions();
-        this.onChange = this.onChange.bind(this);
+        this.controlledOnChange = this.controlledOnChange.bind(this);
+        this.uncontrolledOnChange = this.uncontrolledOnChange.bind(this);
 
         this.state = {
-            currentComponent: testComponents.textField.component,
-            currentComponentSchema: testComponents.textField.schema,
-            currentComponentData: getExample(testComponents.textField.schema),
-            currentComponentOrderByPropertyNames: void(0),
-            currentComponentAttributeAssignment: void(0),
-            onChange: this.onChange,
+            currentControlledComponent: testComponents.textField.component,
+            currentControlledComponentSchema: testComponents.textField.schema,
+            currentControlledComponentData: getExample(testComponents.textField.schema),
+            currentControlledComponentOrderByPropertyNames: void(0),
+            currentControlledComponentAttributeAssignment: void(0),
+            currentUncontrolledComponent: testComponents.textField.component,
+            currentUncontrolledComponentSchema: testComponents.textField.schema,
+            currentUncontrolledComponentData: getExample(testComponents.textField.schema),
+            currentUncontrolledComponentOrderByPropertyNames: void(0),
+            currentUncontrolledComponentAttributeAssignment: void(0),
+            controlledOnChange: this.controlledOnChange,
```

Comment:
It seems a little unconventional to me to store these methods in state - can you speak to why this is necessary?